sherlockml/clients/auth.py

```diff
@@ -46,3 +47,54 @@ def get_access_token(self, client_id, client_secret):
         expires_at = now + timedelta(seconds=body['expires_in'])
 
         return AccessToken(token, expires_at)
+
+
+class SherlockMLAuth(AuthBase):
+    """Requests auth implementation for accessing SherlockML services.
+
+    Parameters
+    ----------
+    auth_service_url : str
+        The URL of the SherlockML authentication service
+    client_id : str
+        The SherlockML client ID to use for authentication
+    client_secret : str
+        The client secret associated with the client ID
+
+    To perform an authenticated request against a SherlockML service, first
+    construct an instance of this class:
+
+    >>> auth = SherlockMLAuth('https://hudson.example.sherlockml.net',
+                              your_client_id, your_client_secret)
+
+    then pass it as the ``auth`` argument when constructing a ``requests``
+    session:
+
+    >>> import requests
+    >>> session = requests.Session(auth=auth)
```

Comment:
When I try this, I get an error:

```python
------------------------------------------------------------------
TypeError                        Traceback (most recent call last)
<ipython-input-11-7e7ea1c88d65> in <module>()
----> 1 requests.Session(auth=authenticator)

TypeError: __init__() got an unexpected keyword argument 'auth'
```

Don't you want to use this syntax from the [requests docs](http://docs.python-requests.org/en/master/user/advanced/):

```python
s = requests.Session()
s.headers.update({'x-test': 'true'})

# both 'x-test' and 'x-test2' are sent
s.get('http://httpbin.org/headers', headers={'x-test2': 'true'})
```

Given that authentication occurs via headers, this would be sufficient, no?

Reply:
(although note this would lose the ability to update the token)

Reply:
I think the syntax for setting the auth mechanism on a session is
```
s = requests.Session()
s.auth = SherlockMLAuth(...)
```

Reply:
yes, this works for me

Reply:
Hmm, I was convinced you could pass auth as a keyword argument at construction time. Will fix this.

Reply:
Done.